cmd/internal-ip/main.go

```diff
@@ -1,63 +1,87 @@
 package main
 
 import (
+	"errors"
+	"flag"
 	"fmt"
 	"net"
 	"os"
 	"strings"
 )
 
+// Set display texts to defaults.
+var fullText = "unknown"
+var shortText = "unknown"
+
 func main() {
+	var iface = &net.Interface{}
+
+	// Check if interface parameter was passed in
+	var ifaceNameFlag = flag.String("ifaceName", "", "-ifaceName wlp2s0")
+	flag.Parse()
+
+	if ifaceNameFlag == nil || *ifaceNameFlag == "" {
+		// ifaceName parameter was not passed, use first interface that is up
+		desiredIface, err := firstNonLoopbackInterfaceThatIsUp()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%s", err.Error())
+			writeAndExit()
+		}
+		iface = desiredIface
+	} else {
+		// ifaceName was passed, use the name to get interface
+		desiredIface, err := net.InterfaceByName(*ifaceNameFlag)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "[i3blocks-go] Failed to find interface with name %s. %s", *ifaceNameFlag, err.Error())
+			writeAndExit()
+		}
+		iface = desiredIface
+	}
+
+	// Retrieve all addresses belonging to
+	// found active interface.
+	addrs, err := iface.Addrs()
+	if err != nil || len(addrs) == 0 {
+		fmt.Fprintf(os.Stderr, "[i3blocks-go] Failed to retrieve IP addresses associated with %s: %v", iface.Name, err)
+		writeAndExit()
+	}
+
+	// Extract the actual IP from address string
+	// and remove subnet information.
+	ipRaw := strings.Split(addrs[0].String(), "/")
 
-	// Set display texts to defaults.
-	var fullText string = "unknown"
-	var shortText string = "unknown"
+	fullText = ipRaw[0]
+	shortText = ipRaw[0]
+	writeAndExit()
+}
 
+func firstNonLoopbackInterfaceThatIsUp() (*net.Interface, error) {
 	// Get all interfaces associated with this machine.
 	ifaces, err := net.Interfaces()
 	if err != nil {
-
 		// Write an error to STDERR, fallback display values
-		// to STDOUT and exit with failure code.
-		fmt.Fprintf(os.Stderr, "[i3blocks-go] Failed to retrieve local interfaces: %s", err.Error())
-		fmt.Fprintf(os.Stdout, "%s\n%s\n", fullText, shortText)
-		os.Exit(0)
+		// to STDOUT and writeAndExit with failure code.
+		return &net.Interface{}, fmt.Errorf("[i3blocks-go] Failed to retrieve local interfaces: %s", err.Error())
```

Comment:
```suggestion
		return nil, fmt.Errorf("[i3blocks-go] Failed to retrieve local interfaces: %s", err.Error())
```

Reply:
```nil``` should be used when you return from a function and want to signal that one pointer value is "unset". ```&net.Interface{}``` will still be set to an "empty" ```net.Interface``` that can be accessed.